app/controllers/api/actions_controller.rb

```diff
@@ -15,12 +15,18 @@ def create
         .merge(mobile_value)
         .merge(extra_fields)
 
-      if action.is_a?(PendingAction)
+      if action.is_a?(PendingActionService)
         path = confirmation_page_path(page, double_opt_in: true,
                                             name: action.data['name'],
                                             email: action.data['email'])
 
-        render js: "location.href = '#{path}';", status: 200
+        action.send_email(version: params[:test_version] || 1)
+
+        if params[:test_version] == '2'
+          render js: 'champaign.DoubleOptIn.showNotice();', status: 200
```

Comment:
This creates a nasty coupling between our javascript bundles and this particular API endpoint.
```rb
render js: 'champaign.DoubleOptIn.showNotice();', status: 200
```
Refactoring the JavaScript side of this is likely to break this endpoint, unless the person refactoring knows and/or remembers to change this. We could add a comment where `DoubleOptIn.showNotice()` is declared, but I'm still weary of adding this type of coupling as there are other ways to do this (that don't add technical debt).